### PR TITLE
[AN-8425] Collect Hadoop metrics after workflow completes

### PR DESCRIPTION
**Pull request by tobz**
We've added a new Ansible playbook (playfile, really) that will execute the collection script for us, providing all the necessary bits to SSH to the proper node.

We've also added a new Make target, which invokes said Ansible playbook, and we now run it after a workflow completes (after `remote-task` returns), before the cluster is terminated.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,6 +14,9 @@ provision.emr: deps
 terminate.emr: deps
 	ansible-playbook --connection local -i 'localhost,' batch/terminate.yml -e "$$EXTRA_VARS"
 
+collect.metrics: deps inventory.refresh
+	ansible-playbook batch/collect.yml -e "$$EXTRA_VARS"
```
Comment (tobz) [inline Makefile]: We specifically omit the localhost/local connection stuff here because we are executing this remotely.

File: batch/collect.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+
+- name: Collect Hadoop metrics
+  hosts: "mr_{{ name }}_master"
+  gather_facts: False
+  vars:
+    - analytics_root: /var/lib/analytics-tasks/automation
+  tasks:
+    - name: run script
+      command: bin/python {{ analytics_root }}/repo/scripts/collect-hadoop-metrics.py {{ analytics_root }}/config/analytics-tasks/collect-hadoop-metrics.yaml
```
Comment (tobz) [inline batch/collect.yml]: Being chdir'd into the venv and using `bin/python` explicitly is what gets us into the virtualenv.  This was the simplest approach I could find for running a Python script within a venv via Ansible.

File: batch/collect.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+
+- name: Collect Hadoop metrics
+  hosts: "mr_{{ name }}_master"
```
Comment (tobz) [inline batch/collect.yml]: `name` comes from `EXTRA_VARS`, being the cluster name.  The special `mr_xxxx_master` portion comes from our tweaked `plugins/ec2.py` inventory script.

File: batch/collect.yml (hunk 2 above)
Comment (mulby) [inline batch/collect.yml]: Is there a way to pretty-print the output from this? So that it appears in the jenkins job console log?

Comment (mulby) [inline batch/collect.yml]: Also, would it be simpler to do:

`{{ analytics_root }}/venv/bin/python {{ analytics_root }}/repo/scripts/collect-hadoop-metrics.py {{ analytics_root }}/config/analytics-tasks/collect-hadoop-metrics.yaml`

Without the chdir?

File: batch/collect.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  gather_facts: False
+  vars:
+    - analytics_root: /var/lib/analytics-tasks/automation
```
Comment (mulby) [inline batch/collect.yml]: I would a comment in here about the hard-code...

Note that the "automation" bit of this needs to match the hard-coded value here too: https://github.com/edx/edx-analytics-configuration/pull/42/files#diff-98ced18f9a4211ac2b6fe736709279e2R41

Comment (tobz) [inline batch/collect.yml]: Hmm, I'm not sure I understand what you mean.

File: batch/collect.yml (hunk 2 above)
Comment (tobz) [inline batch/collect.yml]: Pretty-print which output?  The Ansible output itself is going to be present already, but it seems to truncate the newlines, or turn them into tabs, or something.. basically it doesn't respect the newlines of the command's output.  At least it didn't in my terminal.

And yeah, I could drop the `chdir` bit.

Comment (tobz) [inline batch/collect.yml]: Actually, I ran this stuff with `-vvvv`, so Ansible might NOT display the stdout of the command by default.  I'll look into that.

File: batch/collect.yml (hunk 3 above)
Comment (brianhw) [inline batch/collect.yml]: Seems like some of this might be worth including in a comment.

File: batch/collect.yml (hunk 4 above)
Comment (brianhw) [inline batch/collect.yml]: When run-automated-task calls remote task, it passes "--remote-name automation ".  If we really wanted, the value could be passed by run-automated-task into its call to "make collect.metrics" and through to here.

Comment (tobz) [inline batch/collect.yml]: Right, but what's the relevance?  Why do we expect that value to change?  Compared to the other hard-code things present here.

Comment (mulby) [inline batch/collect.yml]: I'm not attached to using a shared variable as long as there is a comment pointing out the other location that would need to change if someone were to change this value.

Historically the last part of this path has been dynamic. All of the other paths here are very static, but this last part of this path has been a "variable" that was really only hard-coded in run-automated-task.sh. That's the only reason I would treat it differently than the rest of the path (for example).

Comment (brianhw) [inline batch/collect.yml]: Hence a comment.

File: batch/collect.yml (hunk 2 above)
Comment (brianhw) [inline batch/collect.yml]: +1 on the no-chdir way.

**Comment by tobz** [comment]
This should address all of the concerns.

**Pull request merged**